cabal-install/Distribution/Solver/Modular/IndexConversion.hs

```diff
@@ -186,8 +187,36 @@ convGPD os arch cinfo strfl sexes pi
             (L.map  (\(nm, ds) -> conv (ComponentBench nm) benchmarkBuildInfo ds) benchs)
        ++ maybe []    (convSetupBuildInfo pi)    (setupBuildInfo pkg)
 
+    -- | We infer the maximally supported spec-version from @lib:Cabal@'s version
```

Comment:
IIRC haddock comment for non-top things breaks stuff.

Reply:
@phadej tbh, that's a bit vague... can you be more specific?  :-)

Reply:
I think it only affects old versions of Haddock... our CI runs Haddock for cabal-install, so it should catch that.